shell/platform/darwin/ios/framework/Source/FlutterEngine.mm

```diff
@@ -233,6 +236,15 @@ - (void)resetChannels {
 // Channels get a reference to the engine, and therefore need manual
 // cleanup for proper collection.
 - (void)setupChannels {
+  __block FlutterEngine* engine = self;
+  [_binaryMessenger setMessageHandlerOnChannel:@"flutter/isolate"
+                          binaryMessageHandler:^(NSData* message, FlutterBinaryReply reply) {
+                            if (engine) {
+                              engine.isolateId =
+                                  [[[FlutterStringCodec sharedInstance] decode:message] copy];
```

Comment:
Set the isolateId property to copy, then you don't need to call copy on set.

Properties with mutable variants (NSArray, NSString, etc) should always be copy so the object isn't mutated out from under it.

Reply:
Done